src/utils.ts

```diff
@@ -86,6 +90,22 @@ export function toJSONPathArray(jsonPath: string): Array<string | number> {
   return jsonPath.split('/').map(untilde);
 }
 
+export function createUncaghtDiagnostic(err: unknown, message: string, document?: Document): Diagnostic[] {
+  if (err instanceof Error) {
+    const range: Diagnostic['range'] = document ? document.getRangeForJsonPath([]) as Diagnostic['range'] : Document.DEFAULT_RANGE;
+    return [
+      {
+        code: 'uncaught-error',
+        message: `${message}. Name: ${err.name}, message: ${err.message}, stack: ${err.stack}`,
+        path: [],
+        severity: DiagnosticSeverity.Error,
+        range,
+      }
+    ];
+  }
+  return [];
+}
+
```

Comment:
Super minor but readability improvement in my opinion:

```suggestion
export function createUncaghtDiagnostic(err: unknown, message: string, document?: Document): Diagnostic[] {
  if (!err instanceof Error) {
    return [];
  }
  
  const range: Diagnostic['range'] = document ? document.getRangeForJsonPath([]) as Diagnostic['range'] : Document.DEFAULT_RANGE;
  return [
    {
      code: 'uncaught-error',
      message: `${message}. Name: ${err.name}, message: ${err.message}, stack: ${err.stack}`,
      path: [],
      severity: DiagnosticSeverity.Error,
      range,
    }
  ];  
}
```